server/src/controllers/Auth.controller.js

```diff
@@ -68,4 +69,59 @@ export default class AuthController {
       message: 'User registered successfully'
     });
   }
+
+  /**
+   * @method login
+   *
+   * @param {object} req request
+   * @param {object} res response
+   *
+   * @returns {object} returns status code, data and message properties
+   */
+  static login(req, res) {
+    const { email, password } = req.body;
+
+    for (let i = 0; i < users.length; i++) {
+      if (email === users[i].email) {
+        const verifyPassword = comparePassword(password, users[i].password);
+
+        if (!verifyPassword) {
+          return res.status(404).json({
+            status: 404,
+            error: 'Email or password is incorrect'
+          });
+        }
+        const userInfo = users[i];
+
+        const payload = {
+          id: userInfo.id,
+          firstName: userInfo.firstName,
+          lastName: userInfo.lastName,
+          email: userInfo.email,
+          type: userInfo.type,
+          isAdmin: userInfo.isAdmin
+        };
+
+        const token = jwt.sign(payload, process.env.SECRET_KEY, { expiresIn: '1day' });
+
+        const data = {
+          token,
+          id: userInfo.id,
+          firstName: userInfo.firstName,
+          lastName: userInfo.lastName,
+          email: userInfo.email,
+          type: userInfo.type
+        };
+        return res.status(200).json({
+          status: 200,
+          data,
+          message: 'Login successful'
+        });
+      }
+    }
+    return res.status(404).json({
```

Comment:
There are different schools of thought concerning what status code to return to users who are not on the platform at all. Some people choose 404, I usually choose 401 because I believe hackers don't have a right to even know if a user exists on the platform but in some cases, 404 makes more sense e.g platform like Instagram, you want the users to know they don't exist yet. You can research and then choose what you want. You must have your reasons for choosing whatever status code you use.

Reply:
I understand I think it should be 401 since its a banking application, talking about security

Reply:
Okay